versiondb/README.md

```diff
@@ -0,0 +1,141 @@
+
+### Extract Change Sets
+
+```
+$ cronosd changeset dump --home /chain/.cronosd/ --output data distribution acc authz bank capability cronos evidence evm feegrant feeibc feemarket gov ibc mint params slashing staking transfer upgrade
+```
+
+`dump` command will extract the change sets from the IAVL tree, and store each store in separate directories. The change set files are segmented into different chunks and compressed with zlib level 6 by default, the chunk size defaults to 1m blocks, the result `data` directly will look like this:
+
+```
+data/acc/block-0.zz
+data/acc/block-1000000.zz
+data/acc/block-2000000.zz
+...
+data/authz/block-0.zz
+data/authz/block-1000000.zz
+data/authz/block-2000000.zz
+...
+```
+
+Extraction is the slowest step, the test run on testnet archive node takes around 11 hours on a 8core ssd machine, but fortunately, the change set files can be verified pretty fast, so they can be share on CDN in a trustless manner, normal users should just download them from CDN and verify the correctness locally, should be much faster than extract by yourself.
+
+#### Verify Change Sets
+
+```
+$ cronosd changeset verify data/acc/*.zz
+7130689 8DF52D6F7A7690916894AF67B07D64B678FB686626B2B3109813BBE172E74F08
+```
+
+`verify` command will replay all the change sets and rebuild the final IAVL tree and output the root hash of the final version, user can check the root hash against the IAVL tree.
+
+> TODO: will provide command to generate app-hash directly, combining the root hashes of each store, so it'll be easier for end user to verify.
+
+`verify` command takes several minutes and several gigabytes of ram to run, if ram usage is a problem, it can also run incrementally, you can export the snapshot for a middle version, then verify the remaining versions start from that snapshot:
+
+```
+$ cronosd changeset verify --save-snapshot /tmp/snapshot data/acc/block-0.zz data/acc/block-1000000.zz data/acc/block-2000000.zz
+$ cronosd changeset verify --load-snapshot /tmp/snapshot data/acc/block-3000000.zz data/acc/block-4000000.zz data/acc/block-5000000.zz
+```
+
+The format of change set files are documented [here](memiavl/README.md#change-set-file).
+
+### Convert To VersionDB
+
+#### SST File Writing
+
+To maximize the speed of initial data ingestion into rocksdb, we take advantage of the sst file writer in rocksdb, with that we can write out sst files directly without causing contention on a shared database, the sst files for each store can be written out in parallel. We also developed an external sorting algorithm to sort the data before writing the sst files, so the sst files don't cause extra compaction in the final db.
+
+```
+$ cronosd changeset convert-to-sst --store distribution ./sst/distribution.sst data/distribution/*.zz
+```
+
+`convert-to-sst` will do the sst file writing for a single store, you can wrap it in a simple script to run multiple stores in parallel. Here's an example Python script:
```

Comment:
maybe put it directly into `scripts` folder. Or just do it in `convert-to-sst` cli directly, for better UX. In any case, we only convert a few modules instead all?

Reply:
normally user should convert all modules, but yes, technically it's possible to only convert a subset of modules.

Reply:
https://github.com/crypto-org-chain/cronos/issues/826
Opened a separate issue to improve cmd UX.